lib/Sema/TypeCheckCodeCompletion.cpp

```diff
@@ -1127,22 +1127,25 @@ static Type getTypeForCompletion(const constraints::Solution &S, Expr *E) {
     });
 
     return S.simplifyType(completionTy.transform([&](Type type) {
-      if (auto *placeholder = type->getAs<PlaceholderType>()) {
-        if (auto *typeVar =
-                placeholder->getOriginator().dyn_cast<TypeVariableType *>()) {
-          if (auto *GP = typeVar->getImpl().getGenericParameter()) {
-            // Code completion depends on generic parameter type being
-            // represented in terms of `ArchetypeType` since it's easy
-            // to extract protocol requirements from it.
-            if (auto *GPD = GP->getDecl())
-              return GPD->getInnermostDeclContext()->mapTypeIntoContext(GP);
+      auto *placeholder = type->getAs<PlaceholderType>();
+      if (!placeholder)
+        return type;
+
+      if (auto *typeVar =
+              placeholder->getOriginator().dyn_cast<TypeVariableType *>()) {
+        if (auto *GP = typeVar->getImpl().getGenericParameter()) {
+          // Code completion depends on generic parameter type being
+          // represented in terms of `ArchetypeType` since it's easy
+          // to extract protocol requirements from it.
+          if (auto *GPD = GP->getDecl()) {
+            DeclContext *DC = GPD->getInnermostDeclContext();
+            if (auto mappedTy = DC->mapTypeIntoContext(GP))
```

Comment:
I don't think it's possible for mapTypeIntoContext() to return nullptr, since it just calls subst(). A lot of other things would break if subst() started to return nullptr.